File: leetcode/45.jump-game-ii/ts/jump.ts
const jump = (nums: number[]): number => {
    if (nums.length < 2) return 0

    let res = 0
    let curr = 0
    let next = 0

    for (let i = 0; i < nums.length; i++) {
        if (i + nums[i] > next) next = i + nums[i]

        if (i !== curr) continue

        res++
        curr = next
        if (curr >= nums.length - 1) break
    }

    return res
}

File: leetcode/45.jump-game-ii/cpp/jump.cpp
#include <vector>

class Solution
{
public:
    int jump(std::vector<int> &nums)
    {
        if (nums.size() < 2)
            return 0;

        int res = 0;
        int curr = 0;
        int next = 0;

        for (int i = 0; i < nums.size(); i++)
        {
            next = i + nums[i] > next ? i + nums[i] : next;

            if (i != curr)
                continue;

            res++;
            curr = next;
            if (curr >= nums.size() - 1)
                break;
        }

        return res;
    }
};
